src/napari_experimental/group_layer_actions.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List
+
+from app_model.types import Action
+from qtpy.QtCore import QPoint
+from qtpy.QtWidgets import QAction, QMenu
+
+from napari_experimental.group_layer import GroupLayer
+
+if TYPE_CHECKING:
+    from qtpy.QtWidgets import QWidget
+
+
+class GroupLayerActions:
+    """Class holding all GroupLayerActions to be shown in the right click
+    context menu. Based on structure in napari/layers/_layer_actions and
+    napari/app_model/actions/_layerlist_context_actions
+
+    Parameters
+    ----------
+    group_layers: GroupLayer
+        Group layers to apply actions to
+    """
+
+    def __init__(self, group_layers: GroupLayer) -> None:
+        self.group_layers = group_layers
+
+        self.actions: List[Action] = [
+            Action(
+                id="napari:grouplayer:toggle_visibility",
+                title="toggle_visibility",
+                callback=self._toggle_visibility,
+            )
+        ]
+
+    def _toggle_visibility(self):
+        """Toggle the visibility of all selected layers inside the
+        group layers"""
+        for item in self.group_layers.selection:
+            if not item.is_group():
+                visibility = item.layer.visible
+                item.layer.visible = not visibility
```

Comment:
```suggestion
        for node in [item for item in self.group_layers.selection if not item.is_group()]:
            visibility = node.layer.visible
            node.layer.visible = not visibility
```

Personally I've been going with this when I only want to iterate though either nodes or groups - "item" is generic and can refer to either nodes or groups but here we really are just using the nodes.

Reply:
Good point! Although now I've merged the `km/group-vis-toggle` branch, it needs to iterate through both groups and nodes to set visibility on both